File: CS/IPhoneCheckEditSample/Form1.cs
using System.ComponentModel;
using DevExpress.Skins;
using DevExpress.XtraEditors;

namespace IPhoneCheckEditSample {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            InitGrid();
            InitSkinList();      
        }

        void InitGrid() {
            BindingList<GridContentElement> gridContentElements = new BindingList<GridContentElement>();
            gridContentElements.Add(new GridContentElement() { Caption = "Mail", Value = true });
            gridContentElements.Add(new GridContentElement() { Caption = "Contacts", Value = false });
            gridContentElements.Add(new GridContentElement() { Caption = "Calendars", Value = false });
            gridContentElements.Add(new GridContentElement() { Caption = "Bookmarks", Value = false });
            gridControl1.DataSource = gridContentElements;
            gridView1.Columns["Value"].ColumnEdit = repositoryItemIPhoneCheckEdit1;
        }
        void InitSkinList() {
            listBoxControl1.DataSource = SkinManager.Default.Skins;
            listBoxControl1.DisplayMember = "SkinName";
        }
        void listBoxControl1_SelectedIndexChanged(object sender, System.EventArgs e) {
            if((sender as ListBoxControl).SelectedItem == null) return;
            defaultLookAndFeel1.LookAndFeel.SkinName = ((sender as ListBoxControl).SelectedItem as SkinContainer).SkinName;
        }
    }

    class GridContentElement : INotifyPropertyChanged {
        string caption;
        bool value;
        
        public string Caption {
            get { return caption; }
            set {
                if(caption == value) return;
                this.caption = value;
                DoPropertyChanged("Caption");
            }
        }
        public bool Value {
            get { return value; }
            set {
                if(this.value == value) return;
                this.value = value;
                DoPropertyChanged("Value");
            }
        }
        public event PropertyChangedEventHandler PropertyChanged;

        void DoPropertyChanged(string propertyName) {
            if(PropertyChanged == null) return;
            PropertyChanged(this, new PropertyChangedEventArgs(propertyName));
        }
    }
}

File: VB/IPhoneCheckEditSample/Form1.vb
Imports System.ComponentModel
Imports DevExpress.Skins
Imports DevExpress.XtraEditors

Namespace IPhoneCheckEditSample
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
			InitGrid()
			InitSkinList()
		End Sub

		Private Sub InitGrid()
			Dim gridContentElements As New BindingList(Of GridContentElement)()
			gridContentElements.Add(New GridContentElement() With {.Caption = "Mail", .Value = True})
			gridContentElements.Add(New GridContentElement() With {.Caption = "Contacts", .Value = False})
			gridContentElements.Add(New GridContentElement() With {.Caption = "Calendars", .Value = False})
			gridContentElements.Add(New GridContentElement() With {.Caption = "Bookmarks", .Value = False})
			gridControl1.DataSource = gridContentElements
			gridView1.Columns("Value").ColumnEdit = repositoryItemIPhoneCheckEdit1
		End Sub
		Private Sub InitSkinList()
			listBoxControl1.DataSource = SkinManager.Default.Skins
			listBoxControl1.DisplayMember = "SkinName"
		End Sub
		Private Sub listBoxControl1_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles listBoxControl1.SelectedIndexChanged
			If (TryCast(sender, ListBoxControl)).SelectedItem Is Nothing Then
				Return
			End If
			defaultLookAndFeel1.LookAndFeel.SkinName = (TryCast((TryCast(sender, ListBoxControl)).SelectedItem, SkinContainer)).SkinName
		End Sub
	End Class

	Friend Class GridContentElement
		Implements INotifyPropertyChanged
		Private caption_Renamed As String
		Private value_Renamed As Boolean

		Public Property Caption() As String
			Get
				Return caption_Renamed
			End Get
			Set(ByVal value As String)
				If caption_Renamed = value Then
					Return
				End If
				Me.caption_Renamed = value
				DoPropertyChanged("Caption")
			End Set
		End Property
		Public Property Value() As Boolean
			Get
				Return value_Renamed
			End Get
			Set(ByVal value As Boolean)
                If Me.value_Renamed = value Then
                    Return
                End If
				Me.value_Renamed = value
				DoPropertyChanged("Value")
			End Set
		End Property
		Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

		Private Sub DoPropertyChanged(ByVal propertyName As String)
			If PropertyChangedEvent Is Nothing Then
				Return
			End If
			RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
		End Sub
	End Class
End Namespace